extensions/guacamole-auth-restrict/src/main/java/org/apache/guacamole/auth/restrict/RestrictionVerificationService.java

```diff
@@ -182,14 +202,21 @@ public static RestrictionType allowedByHostRestrictions(String allowedHostsStrin
         // action if there are any matches.
         List<HostName> allowedHosts = HostRestrictionParser.parseHostList(allowedHostsString);
         for (HostName hostName : allowedHosts) {
+
+            if (hostName == null) {
+                LOGGER.warn("Host name from allow list is null, skipping and moving to the next.");
+                LOGGER.debug("Receive a null value from the list of allowed hosts, skipping the entry.");
+                continue;
+            }
```

Comment:
Same here - I'm not sure it's possible for `parseHostList()` to return a `List<HostName>` containing `null` entries.